libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SalesforceNetworkPlugin.java

```diff
@@ -198,7 +198,7 @@ private RestRequest prepareRestRequest(JSONArray args) throws UnsupportedEncodin
         final JSONObject arg0 = args.optJSONObject(0);
         if (arg0 != null) {
             final RestRequest.RestMethod method = RestRequest.RestMethod.valueOf(arg0.optString(METHOD_KEY));
-            final String endPoint = arg0.optString(END_POINT_KEY);
+            final String endPoint = arg0.optString(END_POINT_KEY, "");
```

Comment:
null value causes NPE at line 235

Reply:
`optString` is supposed to return empty string (NOT `null`) based on documentation [here](https://developer.android.com/reference/org/json/JSONObject?hl=en#optString(java.lang.String)). Are you sure it's coming back with `null`?